docs/source/extending_pywr/extending_pywr_recorders.rst

```diff
@@ -0,0 +1,6 @@
+.. _extending-pywr-recorders:
+
+Extending Pywr with custom Recorders
+------------------------------------
+
+TODO
```

Comment:
I'm not planning to write this as part of this PR as I don't have much experience with it.